### ISSUE TITLE
CMake cannot find pugixml

### ISSUE THREAD

**Issue by kgreenek** [fzi-forschungszentrum-informatik/Lanelet2#91]
I'm building on Ubuntu 18.04 via catkin.

pugixml has been system-installed using the following command:
```
sudo apt install libpugixml-dev
```

When I upgrade Lanelet2 and mrt_cmake_modules to 1.0.0, I start seeing the following error when I try to build with `catkin build`:

```
CMake Error at /home/kevin/ros_ws/install/share/lanelet2_core/cmake/lanelet2_coreAutoDepsConfig.cmake:216 (find_package):
  By not providing "Findpugixml.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "pugixml", but
  CMake did not find one.

  Could not find a package configuration file provided by "pugixml" with any
  of the following names:

    pugixmlConfig.cmake
    pugixml-config.cmake

  Add the installation prefix of "pugixml" to CMAKE_PREFIX_PATH or set
  "pugixml_DIR" to a directory containing one of the above files.  If
  "pugixml" provides a separate development package or SDK, be sure it has
  been installed.
Call Stack (most recent call first):
  /home/kevin/ros_ws/install/share/lanelet2_io/cmake/lanelet2_ioAutoDepsConfig.cmake:272 (_find_dep)
  /home/kevin/ros_ws/install/share/lanelet2_io/cmake/lanelet2_ioConfig.cmake:40 (find_package)
  /home/kevin/ros_ws/install/share/lanelet2_io/cmake/lanelet2_ioConfig.cmake:52 (_init_package_dependencies)
  CMakeLists.txt:14 (find_package)
```

Previously I was using lanelet2 as of commit: `8a158b611403fdbc0b8b49c3edd4d80c819c6adf` and mrt_cmake_modules at commit: `1cfe6038ac9058d798200d1cabff870037d3b394`, and it was working. Downgrading back to those commits fixes the problem as well.

**Comment by kgreenek** [comment]
I found that it works if I add `-DCMAKE_MODULE_PATH=$HOME/ros_ws/src/mrt_cmake_modules/cmake/Modules` to the cmake args with catkin config. So it looks like the problem might be something to do with setting the module path for transitive dependencies.

**Comment by poggenhans** [comment]
Sorry I cannot reproduce your problem. Is there something special about your workspace configuration?

`mrt_cmake_modules` installs the CMake Modules stuff to `<workspace>/install/share/mrt_cmake_modules/cmake/Modules`. When lanelet2_io `find_package`s mrt_cmake_modules, the extras file in `<workspace>/install/share/mrt_cmake_modules/cmake/mrt_cmake_modules-extras.cmake` is included, which in turn sets the `CMAKE_MODULE_PATH` to the Modules folder, where the Findpugixml.cmake file lives.

If mrt_cmake_modules was installed properly by cmake, I cannot see why this should not work. Also, this behaviour has not changed recently.

**Comment by kgreenek** [comment]
There's nothing special about my workspace (that I'm aware of).

Lanelet2 compiles, but when I try to compile my package that depends on Lanelet2, that's when I get the cmake errors. It's possible there's something wrong with my package, but it's odd that it works with the older commit.

I'll try making a minimal example catkin package that just links against Lanelet2, and see if I can reproduce it.

**Comment by poggenhans** [comment]
Ah, I didn't know that the problem occurs when you trying to build your own package against lanelet2.

I was able to reproduce that behaviour a simple cmake file. It happens when the depending project itself does not use `mrt_cmake_modules`. We didn't notice this problem because all our packages use it. I'll look for a way to fix that...

**Comment by kgreenek** [comment]
Excellent, thanks! And thanks for the quick replies.

Based on your comment, I found that adding this line to the top of my cmake file allows it to compile. The mrt_cmake_modules find script must be setting some global state that's implicitly required by lanelet2.

```
find_package(mrt_cmake_modules REQUIRED NO_MODULE)
```

Anyways, that's a much better work-around for now than explicitly adding the module path to the cmake flags.

**Comment by poggenhans** [comment]
Should be fixed in KIT-MRT/mrt_cmake_modules@d0b0a40
